modules/features/profile/src/main/java/au/com/shiftyjelly/pocketcasts/profile/AccountDetailsFragment.kt

```diff
@@ -242,11 +255,48 @@ class AccountDetailsFragment : BaseFragment() {
         val builder = AlertDialog.Builder(themedContext)
         builder.setTitle(getString(LR.string.profile_sign_out))
             .setMessage(getString(LR.string.profile_sign_out_confirm))
-            .setPositiveButton(getString(LR.string.profile_sign_out)) { _, _ -> performSignOut() }
+            .setPositiveButton(getString(LR.string.profile_sign_out)) { _, _ -> clearDataAlert() }
             .setNegativeButton(getString(LR.string.cancel), null)
             .show()
     }
 
+    private fun clearDataAlert() {
+        val context = context ?: return
+        val themedContext = if (Util.isAutomotive(context)) ContextThemeWrapper(context, CR.style.Theme_Car_NoActionBar) else context
+        val builder = AlertDialog.Builder(themedContext)
+        builder.setTitle(getString(LR.string.profile_clear_data_question))
+            .setMessage("Would you also like to clear all data from the app?")
+            .setPositiveButton("Just sign out") { _, _ -> performSignOut() }
+            .setNegativeButton(getString(LR.string.profile_clear_data)) { _, _ ->
+                signOutAndClearData()
+            }
+            .show()
+    }
+
+    private fun signOutAndClearData() {
+        // Sign out first to make sure no data changes get synced
+        userManager.signOut(playbackManager, wasInitiatedByUser = true)
+
+        // Block while clearing data so that we don't return to the app until the users data has been cleared
+        runBlocking(Dispatchers.IO) {
```

Comment:
I'm using `runBlocking` here because I'm concerned about the problems that might occur if the user starts interacting with the app when we're in the middle of deleting all the data.